aws/aws.go

```diff
@@ -54,12 +55,29 @@ const (
 )
 
 func newSession(region string) *session.Session {
+	// If there were no external credentials configured by the caller, use the default credentials
+	if externalcreds.Get() == nil {
+		return session.Must(
+			session.NewSessionWithOptions(
+				session.Options{
+					SharedConfigState: session.SharedConfigEnable,
+					Config: awsgo.Config{
+						Region: awsgo.String(region),
+					},
+				},
+			),
+		)
+	}
+
+	// If external credentials were passed, use them when instantiating a new session
+	externalCreds := externalcreds.Get()
```

Comment:
NIT: you can make this more DRY if you have the `externalcreds.Get` function take a region and return the empty config without credentials when there are no creds.

Reply:
Fixed in 23111ea